woocommerce/woocommerce
# Show correct variations count when generating a single variation

### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:



This PR corrects the variations count shown when using the "Generate variations" button and a single variation is generated.

#### Before

<img width="795" alt="Screenshot 2023-04-20 at 08 47 03" src="https://user-images.githubusercontent.com/2098816/233370751-54e3ef1e-19ae-4810-8831-a31cfdf8b108.png">

#### After

<img width="795" alt="Screenshot 2023-04-20 at 08 48 24" src="https://user-images.githubusercontent.com/2098816/233370774-8be68bd8-f4ee-4476-aa78-fdd412ec0dd5.png">


Closes #37644.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Go to `Products` > `Add New` > `Product data`.
2. Select a `Variable product` and go to the `Attributes` tab.
3. Add a new attribute with only one value/term and save the attribute.
4. Go to the `Variations` tab and click `Generate variations`.
5. Verify that the variations count is correct.
6. Also, use `Add manually` and `Remove` to add variations and verify the count remains correct.
7. Repeat this with an attribute with more than one value/term and verify that when multiple variations are generated, the count is correct.
